waku/node/waku_node.nim

```diff
@@ -1278,21 +1279,26 @@ proc mountPeerExchange*(
   except LPError:
     error "failed to mount wakuPeerExchange", error = getCurrentExceptionMsg()
 
+proc mountPeerExchangeClient*(node: WakuNode) {.async: (raises: []).} =
+  info "mounting waku peer exchange client"
+  if node.wakuPeerExchangeClient.isNil():
+    node.wakuPeerExchangeClient = WakuPeerExchangeClient.new(node.peerManager)
+
 proc fetchPeerExchangePeers*(
```

Comment:
@Ivansete-status it's interesting how `waku_node.nim` is basically an SDK layer that brings usage of various protocols.

It may be wise to start thinking about splitting it, because the file is growing fast in size.

For example, the peer exchange logic could probably be in its own file (under `factory` is fine), and just brought in here if used.